server/container_restore.go

```diff
@@ -331,6 +333,14 @@ func (s *Server) CRImportCheckpoint(
 			}
 		}
 
+		// Support for the Recursive Read-only (RRO) mount requires
+		// the mount to be read-only and the mount propagation to be
+		// set to private.
+		if mount.RecursiveReadOnly {
+			mount.Readonly = true
+			mount.Propagation = types.MountPropagation_PROPAGATION_PRIVATE
+		}
+
```

Comment:
**To the reviewer:**

Hopefully, this is the right thing to do when restoring a container.